src/test/ui/impl-trait/dyn-trait-elided-two-inputs.rs

```diff
@@ -0,0 +1,16 @@
+// Test that we don't get an error with `dyn Bar` in an impl Trait
+// when there are multiple inputs.  The `dyn Bar` should default to `+
+// 'static`. This used to erroneously generate an error (cc #62517).
+//
+// build-pass
+
+trait Foo { type Item: ?Sized; }
+trait Bar { }
+
+impl<T> Foo for T {
+    type Item = dyn Bar;
+}
+
+fn foo(x: &str, y: &str) -> impl Foo<Item = dyn Bar> { () }
```

Comment:
Can you also include the type parameter version (`Foo<Item>`) here also?

Reply:
And as a suggestion to apply... :)

```suggestion
fn foo(x: &str, y: &str) -> impl Foo<Item = dyn Bar> { () }

trait Alpha<Item: ?Sized> {}
trait Object {}

impl<T> Foo<dyn Object> for T {}

fn alpha(x: &str, y: &str) -> impl Alpha<dyn Object> { () }

```